agent/tool-scripts/jmap

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+# -*- mode: shell-script; indent-tabs-mode: t; sh-basic-offset: 8; sh-indentation: 8; sh-indent-for-case-alt: + -*-
+# The following script collect heap used stats for jvm and containerized jvm
+
+script_path=`dirname $0`
+script_name=`basename $0`
+pbench_bin="`cd ${script_path}/..; /bin/pwd`"
+
+# source the base script
+. "$pbench_bin"/base
+
+# Perftool scripts must provide the following functions
+# 1) Install the tool
+# 2) Start data collection
+# 3) Stop data collection
+# 4) post-process the data
+
+# Defaults
+tool=jmap
+tool_bin=/usr/bin/$tool
+group=default
+dir=""
+mode=""
+iteration="1"
+options="none"
+jcn=""
+
+function usage() {
+echo "
+USAGE:
+	$script_name: you specified an invalid option
+	The following options are available:
+	--install,			        install this perf tool
+	--start|stop|postprocess	start/stop/post-process the data collection
+	--iteration=int 		    the iteration (required)
+	--group=str		            the perftool group (not-required)
+	--dir=str			        directory to store data collection (required)
+	--pid=int		            a process ID to profile
+	--pattern			        collect jvm stats for any PID which name matches this string
+	--jvm-container-name             collect jvm stats for containerized jvm which name matches this string(container name)
+"
+}
+
+
+# Process options and arguments
+opts=$(getopt -q -o idp --longoptions "dir:,group:,iteration:,pid:,pattern:,start,stop,install,postprocess" -n "getopt.sh" -- "$@");
+if [ $? -ne 0 ]; then
+        usage
+        exit 1
+fi
+
+eval set -- "$opts";
+while true; do
+	case "$1" in
+		--install)
+		mode="install"
+		shift;
+		;;
+		--start)
+		mode="start"
+		shift;
+		;;
+		--stop)
+		mode="stop"
+		shift;
+		;;
+		--postprocess)
+		mode="postprocess"
+		shift;
+		;;
+		-d|--dir)
+		shift;
+		if [ -n "$1" ]; then
+			dir="$1"
+			shift
+		fi
+		;;
+		-g|--group)
+		shift;
+		if [ -n "$1" ]; then
+			group="$1"
+			shift
+		fi
+		;;
+		-i|--iteration)
+		shift;
+		if [ -n "$1" ]; then
+			iteration="$1"
+			shift
+		fi
+		;;
+		--pid)
+		shift;
+		if [ -n "$1" ]; then
+			jvm_pid="$1"
+			shift
+		fi
+		;;
+		--pattern)
+		shift;
+		if [ -n "$1" ]; then
+			pattern="$1"
+			shift
+		fi
+		;;
+		-j|--jvm-container-name)
+		shift;
+		if [ -n "$1" ]; then
+			jcn="$1"
+			shift
+		fi
+		;;
+		--)
+		shift;
+		break;
+		;;
+		*)
+		break;
+	esac
+done
+
+tool_dir="$dir/tools-$group"
+tool_output_dir=$tool_dir/$tool # all tools keep data in their tool specific dir
+case "$mode" in
```

Comment:
The spacing and alignment of this case statement needs a bit of work to clean up.  Can you do that before submitting?

Reply:
ok